vignettes/scoring-multivariate-forecasts.Rmd

```diff
@@ -0,0 +1,168 @@
+```{r}
+score(example_univ_single)
+```
+
+Now, of course, we can also score multiple similar forecasts at the same time. Let's say we're not only interested in Germany, but other countries as well.
+
+```{r}
+example_univ_multi <- example_sample_continuous[
+  target_type == "Cases" &
+    forecast_date == "2021-05-03" &
+    target_end_date == "2021-05-15" &
+    horizon == 2 &
+    model == "EuroCOVIDhub-ensemble"
+]
+example_univ_multi
+```
+
+Now, we have a set of 4 forecasts for 4 different countries, each of them represented by a set of 40 samples from the predictive distribution.
+
+When we score these forecasts, we will get 4 scores, one for each forecast and observed value.
+```{r}
+score(example_univ_multi)
+```
+
+## Multivariate forecasts
+
+Now, instead of treating the four observations as independent, we could also think of them as a single realisation of a draw from the multivariate distribution of COVID-19 cases across several countries.
+
+The corresponding multivariate forecast would similarly specify a predictive distribution for the number of cases across all 4 countries. The samples are then not draws from four independent distributions, but instead samples from a joint multivariate predictive distribution.
+
+In the following, let's assume that our samples were draws from a multivariate distribution all along (we just treated them as independent for the univariate case).
+
+To tell `scoringutils` that we want to treat these as a multivariate forecast, we need to specify the columns that are pooled together to form a single multivariate forecast. We do this via the `joint_across` argument. For example, if we want to pool forecasts across locations and treat them as a single multivariate forecast, we could set `joint_across = c("location", "location_name")` (in our example, the two columns contain essentially the same information - we therefore have to include both in `joint_across` (or could alternatively delete one of them)).
+
+```{r}
+example_multiv <- as_forecast_multivariate_sample(
+  data = example_univ_multi,
+  c("location", "location_name")
+)
+example_multiv
+```
+
+The column `.mv_group_id` is created automatically and represents an identifier for each multivariate forecast. `.mv_group_id` is 1 everywhere, because we only have a single multivariate forecast. When scoring this forecast using an appropriate multivariate scoring function, we will get a single score, even though we have 4 observations, one for each country. (Note that for the purposes of scoring, it doesn't matter that sample ids are still 1-40, repeated 4 times, instead of 1-160. `scoringutils` handles this appropriately.)
+
+```{r}
+score(example_multiv)
+```
+
+If, at any point, you want to score the same forecast using different groupings, you'd have create a new separate forecast object with a different grouping and score that new forecast object.
+
+<!-- ## Details on the grouping - WIP
```

Comment:
Is there an issue for this?

Reply:
I don't think this warrants an issue - to me, this is jus the design choice and expected behaviour. 
We *could* have a function to specify a new `joint_across`, but to me this is just `as_forecast_multivariate_sample`

Reply:
well if there is no future work planned, I don't see why the comment is still in here as WIP?